code/obj/item/device/radios/intercoms.dm

```diff
@@ -26,15 +39,10 @@
 		screen_image.alpha = 180
 		src.UpdateOverlays(screen_image, "screen")
 		if(src.pixel_x == 0 && src.pixel_y == 0)
-			switch(src.dir)
-				if(NORTH)
-					pixel_y = -21
-				if(SOUTH)
-					pixel_y = 24
-				if(EAST)
-					pixel_x = -21
-				if(WEST)
-					pixel_x = 21
+			set_dir(src.dir)
+	mats = 3
+	deconstruct_flags = 102
```

Comment:
Should have the `|`-ed constants here rather than the result of `|`-ing the constants.